File: 백준/Silver/10773. 제로/제로.py
import sys

input = sys.stdin.readline

n = int(input())

ans = 0
stack = []
for _ in range(n):
    num = int(input())
    if num == 0:
        ans = ans - stack.pop()
    else:
        ans = ans + num
        stack.append(num)
        
print(ans)
    

File: 백준/Silver/10773. 제로/제로.cc
#include <iostream>
#include <vector>

using namespace std;

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);
    
    int n;
    cin >> n;
    
    vector<int> stack;
    for(int i = 0; i < n; i++) {
        int num;
        cin >> num;
        
        if (num == 0) {
            stack.pop_back();
        } else {
            stack.push_back(num);
        }
    }
    int ans = 0;
    for(int x:stack) {
        ans = ans + x;
    }
    cout << ans;
    return 0;
}